### PR TITLE
Bump core kit 2.3.1

### PR DESCRIPTION
**Pull request by vigh-m**
**Description of changes:**
Bump to latest core kit release v2.3.1

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

**Review by bcressey — APPROVED** [review]

**Review by KCSesh — APPROVED** [review]

File: Twoliter.lock [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
 
 [sdk]
 name = "bottlerocket-sdk"
-version = "0.42.0"
+version = "0.43.0"
 vendor = "bottlerocket"
-source = "public.ecr.aws/bottlerocket/bottlerocket-sdk:v0.42.0"
-digest = "myHHKE41h9qfeyR6V6HB0BfiLPwj3QEFLUFy4TXcR10="
+source = "public.ecr.aws/bottlerocket/bottlerocket-sdk:v0.43.0"
+digest = "1zKyWC/pXdpzKfNj1Xp1YvEWdyltI7EhAeiNwlikBz8="
```
Comment (koooosh) [inline Twoliter.lock:10]: Should updates to the SDK `source` and  `digest` in the lock file be included in the commit bumping the SDK?

Comment (vigh-m) [reply Twoliter.lock:10]: I think so. The build will fail if you don't update the `.lock` file after updating the `Twoliter.toml`.

Comment (koooosh) [reply Twoliter.lock:10]: Sorry let me rephrase: Currently, the SDK `source` and `digest` updates in the lock file are included in the commit bumping the core-kit version. Should this be moved to the commit bumping the SDK version instead?

I'm not sure if this is an issue, but currently if someone just picks the SDK version update commit, the build will fail -- basically that commit is "non-atomic"

Comment (vigh-m) [reply Twoliter.lock:10]: I see what you mean. Yeah I'm not sure. I think there is some dependency between the core-kit and sdk version which requires both, the main repo and the kit to be on the same sdk version so taking just one of these changes would lead to an undefined state as well.

**Review by koooosh — APPROVED** [review]

**Comment by vigh-m** [comment]
Rebased to use the latest Twoliter version
